### PR TITLE
Javascript Project: Testing Practice

### PR DESCRIPTION
**Pull request by ferbaco86**
On this project we applied the basic knowledge gained on testing units in JavaScript with Jest, by reading the Odin Project Material and the Jest documentation webpage. 

### Project specifications
Write tests for the following functions, and then make the tests pass:

- [x] capitalize(string) takes a string and returns that string with the first character capitalized.

- [x] reverseString(string) takes a string and returns it reversed.

- [x] A calculator object that contains the basic operations: add, subtract, divide, and multiply.

- [x] Caesar Cipher.

- [x] Array Analysis. Write a function that takes an array of numbers and returns an object with the following properties: average, min, max, and length.

### General requirements:

- [x] Add a descriptive README file to your repository

- [x] The README must include instructions on how to set up the project on your machine

- [x] Use descriptive commit messages

- [x] Do not commit console.log

- [x] Do not Window.alert() or Window.confirm() console.log

### GitFlow:

- [x] Create a development branch and set it as a default branch in your repository.

- [x] Create a feature-branch branch (give it a descriptive name)

- [x] Use Linters to validate your code

### CODE REVIEW

File: js/calculator.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  if (typeof a === 'number' && typeof b === 'number') {
+    message = a / b;
+  }
+  return message;
+};
+
+const calc = {
+  add, multiplication, subtract, divide,
+};
+export default calc;
```
Comment (lekegitrepo) [inline js/calculator.js]: #### `calculator` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/calculator_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

**_Although you have all these functions available it should be within an object_**

* [x] It should be an object
* [x] It should have a `add` method
* [x] It should have a `subtract`
* [x] It should have a `multiply` method
* [x] It should have a `divide` method

File: test/caesar.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  expect(caesar.caesarCypher(3, 1)).toBe('The input has to be string');
+});
+
+test('Uppercase letters also works', () => {
+  expect(caesar.caesarCypher('HELLO', 2)).toBe('jgnnq');
+});
+
+test('Punctuation and special characters are not converted', () => {
+  expect(caesar.caesarCypher('h.ello', 2)).toBe('j.gnnq');
+});
```
Comment (lekegitrepo) [inline test/caesar.test.js]: #### Caesar Cipher ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/caesar_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/testing-practice))

* [x] It should have a test for wrapping from `z` to `a`
* [x] It should have a test for keeping the same case
* [x] It should have a test for punctuation

File: js/reverse.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const reverse = (text) => {
```
Comment (lekegitrepo) [inline js/reverse.js]: - **_[Optional] Consider using the name of the function(`reverseString`) specified in the project specification_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Request changes :recycle:
#### Hi @ferbaco86 and @kbjude :clap: :clap:
#### You have done a good job :+1: However, some requirements are missing or were not fully implemented. Check my comment :point_down:

**_Fix these and submit for another review_**

#### _Reviewed by Temitayo Adeleke_

**Review by lucasmazo23 — APPROVED** [review]
## Project approved 🚀
### Great job!
### Your project looks great!
### You can go ahead and merge this pull request!
### Happy coding! 💻
## Lucas Mazo

**Pull request merged**